From: Juan Carlos García Vázquez <gavajc@gmail.com>
Date: Tue, 24 Nov 2020 00:10:29 +0700
Subject: [PATCH] Bozorth: Remove duplicate declaration for stderr

--- libfprint/nbis/include/bozorth.h	2020-02-10 03:20:25.000000000 -0800
+++ libfprint/nbis/include/bozorth.h	2020-11-23 11:40:51.965048500 -0800
@@ -217,8 +217,6 @@
 /**************************************************************************/
 /* Globals supporting command line options */
 extern int verbose_threshold;
-/* Global supporting error reporting */
-extern FILE *stderr;
 
 /**************************************************************************/
 /* In: BZ_GBLS.C */
