From: Juan Carlos García Vázquez <gavajc@gmail.com>
Date: Mon, 26 Apr 2021 00:09:29 +0700
Subject: [PATCH] Capture: Add compatibility with Windows removing Unix signal for standard C signal.

--- examples/img-capture.c	2021-03-10 20:13:07.000000000 -0600
--- examples/img-capture.c	2021-04-26 00:40:58.778851808 -0500
@@ -23,17 +23,18 @@
 #define FP_COMPONENT "example-capture"
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <libfprint/fprint.h>
-#include <glib-unix.h>
 
 #include "storage.h"
 #include "utilities.h"
 
-typedef struct CaptureData
+void *ptrCapture = NULL;
+
+typedef struct _CaptureData
 {
   GMainLoop    *loop;
   GCancellable *cancellable;
-  unsigned int  sigint_handler;
   int           ret_value;
   const char   *filename;
 } CaptureData;
@@ -41,7 +42,6 @@
 static void
 capture_data_free (CaptureData *capture_data)
 {
-  g_clear_handle_id (&capture_data->sigint_handler, g_source_remove);
   g_clear_object (&capture_data->cancellable);
   g_main_loop_unref (capture_data->loop);
   g_free (capture_data);
@@ -96,6 +96,7 @@
     }
 
   save_image_to_pgm (image, capture_data->filename);
+  g_object_unref(image);
 
   capture_quit (dev, capture_data);
 }
@@ -125,14 +126,15 @@
   start_capture (dev, capture_data);
 }
 
-static gboolean
-sigint_cb (void *user_data)
+static void sig_handler(int signo)
 {
-  CaptureData *capture_data = user_data;
-
-  g_cancellable_cancel (capture_data->cancellable);
+    if (signo == SIGINT)
+    {
+        CaptureData *cp_data = (CaptureData *) ptrCapture;
 
-  return G_SOURCE_CONTINUE;
+        printf("Received SIGINT Cancelling\n");
+        g_cancellable_cancel (cp_data->cancellable);
+    }
 }
 
 int
@@ -143,6 +145,8 @@
   GPtrArray *devices;
   FpDevice *dev;
 
+  signal(SIGINT, sig_handler);
+
   setenv ("G_MESSAGES_DEBUG", "all", 0);
   setenv ("LIBUSB_DEBUG", "3", 0);
 
@@ -173,11 +177,9 @@
   capture_data->ret_value = EXIT_FAILURE;
   capture_data->loop = g_main_loop_new (NULL, FALSE);
   capture_data->cancellable = g_cancellable_new ();
-  capture_data->sigint_handler = g_unix_signal_add_full (G_PRIORITY_HIGH,
-                                                         SIGINT,
-                                                         sigint_cb,
-                                                         capture_data,
-                                                         NULL);
+
+  ptrCapture = capture_data;
+
   if (argc == 2)
     capture_data->filename = argv[1];
   else
