From: Juan Carlos García Vázquez <gavajc@gmail.com>
Date: Thu, 02 Jul 2021 09:40:50 +0800
Subject: [PATCH] libfprint-build: Add sources in build file for export bozorth algorithms and enable/disable elanspi.

--- libfprint/meson.build	2021-06-30 08:58:12.000000000 -0500
+++ libfprint/meson.build	2021-07-02 09:36:27.440925007 -0500
@@ -16,9 +16,10 @@
     'fpi-print.c',
     'fpi-ssm.c',
     'fpi-usb-transfer.c',
-    'fpi-spi-transfer.c',
 ]
 
+other_sources = ['fp-boz-algm.c']
+
 libfprint_public_headers = [
     'fp-context.h',
     'fp-device.h',
@@ -27,6 +28,8 @@
     'fp-print.h',
 ]
 
+libfprint_other_headers = ['fp-boz-algm.h']
+
 libfprint_private_headers = [
     'fpi-assembling.h',
     'fpi-byte-reader.h',
@@ -41,7 +44,6 @@
     'fpi-minutiae.h',
     'fpi-print.h',
     'fpi-usb-transfer.h',
-    'fpi-spi-transfer.h',
     'fpi-ssm.h',
 ]
 
@@ -157,9 +159,11 @@
     if driver == 'elan'
         drivers_sources += [ 'drivers/elan.c' ]
     endif
-	if driver == 'elanspi'
-		drivers_sources += [ 'drivers/elanspi.c' ]
-	endif
+    if driver == 'elanspi'
+        drivers_sources += [ 'drivers/elanspi.c' ]
+        libfprint_private_sources += [ 'fpi-spi-transfer.c' ]
+        libfprint_private_headers += [ 'fpi-spi-transfer.h' ]
+    endif
     if driver == 'virtual_image'
         drivers_sources += [ 'drivers/virtual-image.c' ]
     endif
@@ -201,8 +205,6 @@
     drivers_sources += ['drivers/aes3k.c' ]
 endif
 
-other_sources = []
-
 fp_enums = gnome.mkenums_simple('fp-enums',
     sources: libfprint_public_headers,
     install_header: true,
@@ -305,7 +307,7 @@
         gusb_dep,
     ])
 
-install_headers(['fprint.h'] + libfprint_public_headers,
+install_headers(['fprint.h'] + libfprint_public_headers + libfprint_other_headers,
     subdir: versioned_libname
 )
 
