From: Juan Carlos García Vázquez <gavajc@gmail.com>
Date: Tue, 27 Apr 2021 00:10:29 +0700
Subject: [PATCH] Utilities: Add portable wrapper for setenv function

--- examples/utilities.h	2021-03-10 20:13:07.000000000 -0700
+++ examples/utilities.h	2021-04-26 00:13:57.000000000 -0700
@@ -17,9 +17,17 @@
  * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
+#ifndef _UTILITIES_H
+#define _UTILITIES_H
 
 #pragma once
 
 FpDevice * discover_device (GPtrArray *devices);
 FpFinger finger_chooser (void);
 const char * finger_to_string (FpFinger finger);
+
+#ifdef __MSYS__
+int setenv(const char *name, const char *value, int overwrite);
+#endif
+
+#endif

--- examples/utilities.c	2021-03-10 20:13:07.000000000 -0600
--- examples/utilities.c	2021-04-27 10:02:45.000000000 -0500
@@ -22,6 +22,7 @@
 
 #include <libfprint/fprint.h>
 #include <stdio.h>
+#include <stdlib.h>
 
 #include "utilities.h"
 
@@ -125,3 +126,15 @@
 
   return i;
 }
+
+#ifdef __MSYS__
+
+int setenv(const char *name, const char *value, int overwrite)
+{   
+    if (!overwrite)
+        if (getenv(name) != NULL)
+            return 0;
+    
+    return _putenv_s(name, value);
+}
+#endif
